src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -312,59 +340,34 @@ Visitor::visitTypesafe(Parser::GroupGraphPatternSubContext* ctx) {
             varOrTerm(std::move(triple.object_))};
   };
 
-  vector<SparqlTriple> triples = ad_utility::transform(
-      ad_utility::flatten(
-          visitVector<vector<TripleWithPropertyPath>>(ctx->triplesBlock())),
-      convertTriple);
-  auto [ops, filters] =
-      splitUp(visitVector<std::variant<GraphPatternOperation, SparqlFilter>>(
-          ctx->graphPatternNotTriples()));
-  ops.emplace_back(
-      GraphPatternOperation::BasicGraphPattern{std::move(triples)});
-  return {ops, filters};
-}
-
-// ____________________________________________________________________________________
-vector<TripleWithPropertyPath> Visitor::visitTypesafe(
-    Parser::TriplesBlockContext* ctx) {
-  auto triples = visitTypesafe(ctx->triplesSameSubjectPath());
+  BasicGraphPattern triples = {ad_utility::transform(
+      visitTypesafe(ctx->triplesSameSubjectPath()), convertAndRegisterTriple)};
   if (ctx->triplesBlock()) {
-    ad_utility::appendVector(triples, visitTypesafe(ctx->triplesBlock()));
+    triples.appendTriples(visitTypesafe(ctx->triplesBlock()));
   }
-  return triples;
+  return {triples};
```

Comment:
Are those `{}` needed here? They don't help with "explicit" so it should work without them, or am I missing something?